courses/advanced-ada/chapters/generics.rst

```diff
@@ -1084,19 +1086,335 @@ In this example, we declare the array :ada:`A` with elements of both
 we call the :ada:`Average` function from :ada:`Ops`, an instance of the
 generic package :ada:`Float_Interface_Pkg.Ops`.
 
-Discussion: Generic interfaces vs. other approaches
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Discussion: formal interfaces vs. other approaches
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+:code-config:`reset_accumulator=True`
+
+In Ada, we basically have three approaches to describe interfaces. In
```

Comment:
to describe interfaces *for generic types*. Even though we're in a section on generics I think this needs to be really clear.